core/src/main/scala/zio/jdbc/JdbcDecoder.scala

```diff
@@ -27,25 +26,21 @@ import scala.collection.immutable.ListMap
  * a `ResultSet`.
  */
 trait JdbcDecoder[+A] { self =>
-  def unsafeDecode(rs: RowState): (RowState, A)
+  def unsafeDecode(columIndex: Int, rs: ResultSet): (Int, ResultSet, A)
```

Comment:
The `ResultSet` is mutable, do we need to return it? It's the same one that's in the input to the function, so the caller already has access to it.